server/ingredient/model.ts

```diff
@@ -0,0 +1,46 @@
+import type {Types} from 'mongoose';
+import {Schema, model} from 'mongoose';
+
+/**
+ * This file defines the properties stored in an Ingredient
+ * DO NOT implement operations here ---> use collection file
+ */
+
+// Type definition for Ingredient on the backend
+export type Ingredient = {
+  _id: Types.ObjectId; // MongoDB assigns each object this ID on creation
+  name: string;
+  quantity: number;
+  unit: string;
+};
+
+export type PopulatedIngredient = {
+  _id: Types.ObjectId; // MongoDB assigns each object this ID on creation
+  name: string;
+  quantity: number;
+  unit: string;
+};
```

Comment:
You don't need this, since all the types are simple objects like string or number so there is nothing to populate.

Reply:
removed